src/Compilers/CSharp/Portable/Symbols/SymbolExtensions.cs

```diff
@@ -815,5 +815,29 @@ internal static ImmutableArray<INamespaceSymbol> GetPublicSymbols(this Immutable
         {
             return symbol.GetSymbol<FunctionPointerTypeSymbol>();
         }
+
+        /// <summary>
+        /// Returns true if the method has a [AsyncMethodBuilder(typeof(B))] attribute. If so it returns the "B".
+        /// Validation of builder type B is left for elsewhere. This method returns B without validation of any kind.
+        /// </summary>
+        internal static bool HasAsyncMethodBuilder(this Symbol symbol, [NotNullWhen(true)] out object? builderArgument)
+        {
+            Debug.Assert(symbol is not null);
+
+            // Find the AsyncMethodBuilder attribute.
+            foreach (var attr in symbol.GetAttributes())
+            {
+                if (attr.IsTargetAttribute(symbol, AttributeDescription.AsyncMethodBuilderAttribute)
+                    && attr.CommonConstructorArguments.Length == 1
+                    && attr.CommonConstructorArguments[0].Kind == TypedConstantKind.Type)
+                {
+                    builderArgument = attr.CommonConstructorArguments[0].ValueInternal!;
```

Comment:
>ValueInternal [](http://example.com/codeflow?start=73&length=13)

If we know this is a type, why not have the builderArgument be a TypeSymbol?

Reply:
Kept in line with existing `IsCustomTaskType` method.